sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetInteroperabilitySuite.scala

```diff
@@ -96,6 +97,58 @@ class ParquetInteroperabilitySuite extends ParquetCompatibilityTest with SharedS
     }
   }
 
+  test("parquet files with legacy mode and schema evolution") {
+    // This test case writes arrays in Parquet legacy mode and schema evolution and verifies that
+    // the data can be correctly read back.
+
+    Seq(false, true).foreach { legacyMode =>
+      withSQLConf(SQLConf.PARQUET_WRITE_LEGACY_FORMAT.key -> legacyMode.toString) {
```

Comment:
Thank you for adding this test case, @sadikovi . What happens if we write with `spark.sql.parquet.writeLegacyFormat=true` and try to read `spark.sql.parquet.writeLegacyFormat=false` and schema evolution? Do we have a test coverage for that?

Reply:
`spark.sql.parquet.writeLegacyFormat` is only applicable for writes, it is irrelevant when reading a Parquet file.